pkg/scannerv4/client/client.go

```diff
@@ -95,27 +95,41 @@ func (c *gRPCScanner) Close() error {
 }
 
 func createGRPCConn(ctx context.Context, o connOptions) (*grpc.ClientConn, error) {
-	connOpts := []clientconn.ConnectionOption{
-		clientconn.UseInsecureNoTLS(o.skipTLS),
-		clientconn.ServerName(o.serverName),
-		clientconn.MaxMsgReceiveSize(env.ScannerV4MaxRespMsgSize.IntegerSetting()),
-		clientconn.WithDialOptions(
-			// Scanner v4 Indexer and Matcher pods are accessed via gRPC, which Kubernetes does not
-			// load balance too well on its own. We opt to do client-side load balancing, instead,
-			// via DNS name resolution, which is possible because Scanner v4 services are "headless"
-			// (clusterIP: None).
-			//
-			// We just use basic round-robin load balancing at this time.
-			grpc.WithDefaultServiceConfig(`{"loadBalancingConfig": [{"round_robin": {}}]}`),
-		),
-	}
-
 	// Replace HTTP(S) with DNS for client-side load balancing.
 	// This ensures we use the builtin DNS name resolver.
 	address := strings.TrimPrefix(o.address, "https://")
 	address = strings.TrimPrefix(address, "http://")
 	address = "dns:///" + address
```

Comment:
Specifying the protocol in the address should be rejected; address is not an URL, but `[<host>]:<port>`, can you please remove these lines and update?

BTW, Scanner V4 does not offer non-TLS endpoints. We either skip or attempt TLS verification. I understand your confusion, though.

Reply:
> in general for gRPC or for our specific purpose? For gRPC, target is just a valid URI:

For our purposes. The client library assumes Scanner is behind a TCP endpoint, and the actual mechanism/protocol used to communicate with it is an implementation detail. We could even change to something different without having to update the client-side code.

> I can add a validation step for the address specified for the scanner v4 client, if you'd prefer we only support [host]:port

That would be ideal, but since we are the only ones using that interface, it might be OK to move forward without the validation. To me is the stripping of HTTP URL scheme that is confusing.

Reply:
i added validation, but I also decided to keep the http(s):// trimming in case we make some changes and decide to allow the scheme later. This just adds some protection in case we allow that in the future to ensure we properly prefix with `dns:///`

Reply:
I am sorry, but I don't understand this. I get confused by the fact that we have the trimming code there -- it communicates intent and opens up for assumptions. I am afraid having the trimming makes this even more confusing now that we don't even allow the scheme in the string.

Reply:
it's meant as extra protection in case we decide to allow URLs instead of just `[host]:port`. I'll just remove this and maybe add a comment to the validation